mass_mailing_partner/models/mailing_contact.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2015 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2015 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2015 Javier Iniesta <javieria@antiun.com>
+# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
+# Copyright 2020 Tecnativa - Manuel Calero
+# Copyright 2020 Hibou Corp.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class MailingContact(models.Model):
+    _inherit = "mailing.contact"
+
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Partner", domain=[("email", "!=", False)]
+    )
+
+    @api.constrains("partner_id", "list_ids")
+    def _check_partner_id_list_ids(self):
+        for contact in self:
+            if contact.partner_id:
+                other_contact = self.search(
+                    [
+                        ("partner_id", "=", contact.partner_id.id),
+                        ("id", "!=", contact.id),
+                    ]
+                )
+                if contact.list_ids & other_contact.mapped("list_ids"):
+                    raise ValidationError(
+                        _("Partner already exists in one of these mailing lists")
+                        + ": %s" % contact.partner_id.display_name
+                    )
+
+    @api.onchange("partner_id")
+    def _onchange_partner_mass_mailing_partner(self):
+        if self.partner_id:
+            self.name = self.partner_id.name
+            self.email = self.partner_id.email
+            self.title_id = self.partner_id.title
+            self.company_name = self.partner_id.company_id.name
+            self.country_id = self.partner_id.country_id
+            category_ids = self.partner_id.category_id
+            if category_ids:
+                self.tag_ids = category_ids
+
+    @api.model
+    def create(self, vals):
+        record = self.new(vals)
+        if not record.partner_id:
+            record._set_partner()
+        record._onchange_partner_mass_mailing_partner()
+        new_vals = record._convert_to_write(record._cache)
+        new_vals.update(
+            subscription_list_ids=vals.get("subscription_list_ids", []),
+            list_ids=vals.get("list_ids", []),
+        )
+        return super(MailingContact, self).create(new_vals)
+
+    def write(self, vals):
+        for contact in self:
+            new_vals = contact.copy_data(vals)[0]
+            record = self.new(new_vals)
+            if not record.partner_id:
+                record._set_partner()
+            record._onchange_partner_mass_mailing_partner()
+            new_vals = record._convert_to_write(record._cache)
+            new_vals.update(
+                subscription_list_ids=vals.get("subscription_list_ids", []),
+                list_ids=vals.get("list_ids", []),
+            )
+            super(MailingContact, contact).write(new_vals)
```

Comment:
```suggestion
            super().write(new_vals)
```

Reply:
@victoralmau Thanks for your review.

This `super` here inside a for-loop and I don't think it should be changed.